File: backend/src/main/java/com/romeo/VetLink/shopItem/domain/ShopItem.java
package com.romeo.VetLink.shopItem.domain;

import com.romeo.VetLink.config.OwnedEntity;
import com.romeo.VetLink.subscription.domain.Subscription;
import com.romeo.VetLink.user.User;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.security.core.context.SecurityContextHolder;

import java.util.List;
import java.util.Set;

@EqualsAndHashCode(callSuper = true)
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "shop_item")
public class ShopItem extends OwnedEntity {
    @Id
    @GeneratedValue
    private Integer id;

    private String name;
    private String category;
    private String price;
    private String quantity;
    private String shortDescription;

    @ManyToMany(mappedBy = "shopItems", fetch = FetchType.EAGER)
    private List<Subscription> subscriptions;

    @Override
    public void prePersist(){
        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
        this.owner = user.getClinicId();
    }


}

File: mobile/lib/model/shop-item.dart


class ShopItem {
  final String name;
  final String category;
  final String price;
  final String shortDescription;
  final String quantity;
  final int owner;
  final int id;


  const ShopItem({
    required this.name,
    required this.category,
    required this.price,
    required this.shortDescription,
    required this.quantity,
    required this.owner,
    required this.id,
  });

  static ShopItem fromMap(Map<String, dynamic> map) {
    return ShopItem(
      name: map['name'],
      category: map['category'],
      price: map['price'],
      shortDescription: map['shortDescription'],
      quantity: map['quantity'],
      owner: map['owner'],
      id: map['id'],
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'name': name,
      'price': price,
      'category': category,
      'shortDescription': shortDescription,
      'quantity': quantity,
      'owner': owner,
      'id': id
    };
  }
}
